test/e2e/volumes.go

```diff
@@ -340,6 +340,16 @@ func deleteCinderVolume(name string) error {
 	return err
 }
 
+func deletePDVolume(name string) error {
+	output, err := exec.Command("gcloud", "compute", "disks", "delete", name, "-q", "--zone", framework.TestContext.CloudConfig.Zone).CombinedOutput()
```

Comment:
Use the GCE API, it is far more robust. See https://github.com/kubernetes/kubernetes/blob/master/test/e2e/pd.go#L323. Factor out the `createPD()` code from their to share if it makes sense.